junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -129,12 +129,17 @@ public void beforeEach(ExtensionContext context) {
 	}
 
 	private static void installFailureTracker(ExtensionContext context) {
-		context.getStore(NAMESPACE).put(FAILURE_TRACKER, (CloseableResource) () -> context.getParent() //
-				.ifPresent(parentContext -> {
-					if (selfOrChildFailed(context)) {
-						parentContext.getStore(NAMESPACE).put(CHILD_FAILED, true);
-					}
-				}));
+		context.getParent() //
+				.filter(parentContext -> !context.getRoot().equals(parentContext)) //
```

Comment:
:speech_balloon: This line is just an optimization because there never are temp dirs on the engine level so we don't need to store the information there because it will never be read.

Reply:
Unrelated and potentially breaking - but should be okay.